File: sdk/typescript/src/api/referral.ts
import type { HttpClient, RequestOptions } from "../http.js";

export interface ClaimReferralBody {
  /** Referral code extracted from the referral link. */
  code: string;
  /** Optional device fingerprint for abuse prevention. */
  deviceFingerprint?: string;
}

/**
 * Referral link claiming and earnings/referral-list statistics.
 */
export class ReferralApi {
  constructor(private readonly http: HttpClient) {}

  /** Claim a referral link for the authenticated user. */
  claimReferral<T = unknown>(
    body: ClaimReferralBody,
    options?: RequestOptions,
  ): Promise<T> {
    return this.http.post<T>("/referral/claim", body, options);
  }

  /** Fetch referral link, earnings summary, and referral list. */
  getReferralStats<T = unknown>(options?: RequestOptions): Promise<T> {
    return this.http.get<T>("/referral/stats", options);
  }
}

File: sdk/python/src/tinyhumans/api/referral.py
from __future__ import annotations

from typing import Any

from ..http import Json
from ._base import ApiNamespace

__all__ = ["ReferralApi"]


class ReferralApi(ApiNamespace):
    """Referral link claiming and earnings/referral-list statistics."""

    def claim_referral(self, body: dict[str, Any], **kwargs: Any) -> Json:
        """Claim a referral link for the authenticated user."""
        return self._http.post("/referral/claim", body=body, **kwargs)

    def get_referral_stats(self, **kwargs: Any) -> Json:
        """Fetch referral link, earnings summary, and referral list."""
        return self._http.get("/referral/stats", **kwargs)
